### PR TITLE
MM-48423 : Landing on "Team not found" page on clean slate browser

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
Back when delete_at was not part of gQL query, we hardcoded it with `delete_at = 0`, after the support was added, we never changed that. This pr adds that and other fields to team object. Also adds an e2e test.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48423

#### Related Pull Requests
n/a

#### Screenshots
n/a

#### Release Note

```release-note
Fixed bug in webapp in graphQL environment where it became possible to for user to visit archived team.
```

### CODE REVIEW

**Review by m1lt0n — APPROVED** [review]

**Comment by M-ZubairAhmed** [comment]
> Do we need to bind this change to a minimum server version? When was this added to the server gql response?

No i dont think so, since its been a long time this was added in https://github.com/mattermost/mattermost-server/pull/20610

File: e2e/cypress/tests/integration/team/archive_team_spec.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+describe('archive team', () => {
+    let adminUser: Cypress.UserProfile;
+
+    before(() => {
+        cy.apiCreateCustomAdmin({loginAfter: true}).then(({sysadmin}) => {
+            adminUser = sysadmin;
+        });
+    });
+
+    it('MM-XXXX should not go to deleted team on login', () => {
```
Comment (M-ZubairAhmed) [inline e2e/cypress/tests/integration/team/archive_team_spec.ts:15]: @furqanmlk this will need the test id

**Comment by M-ZubairAhmed** [comment]
e2e not 100% due to some flaky tests. running again

**Comment by M-ZubairAhmed** [comment]
e2e report : https://automation-dashboard.vercel.app/cycles/98f95ae3-09ce-4a7e-84fc-9edb27046c25

**Review by agnivade — APPROVED** [review]

**Pull request merged**